src/transformers/pipelines/text2text_generation.py

```diff
@@ -230,6 +242,22 @@ class TranslationPipeline(Text2TextGenerationPipeline):
 
     # Used in the return key of the pipeline.
     return_name = "translation"
+    src_lang: Optional[str] = None
+    tgt_lang: Optional[str] = None
+
+    def __init__(self, *args, src_lang=None, tgt_lang=None, **kwargs):
+        super().__init__(*args, **kwargs)
+        if src_lang is not None:
+            self.src_lang = src_lang
+        if tgt_lang is not None:
+            self.tgt_lang = tgt_lang
+        if src_lang is None and tgt_lang is None:
+            task = kwargs.get("task", "")
+            tokens = task.split("_")
+            if task and len(tokens) == 4:
```

Comment:
maybe a else `raise ValueError(....)`

Reply:
I don't think we should treat as `invalid` any other task argument here.
We are already within the `TranslationPipeline` so the actual task is  implied no matter the argument.
We are merely trying to backport `task=translation_xx_to_yy` when it's easy (.

But if code was written with `task=translation_en_XX_to_ar_AR` for instance, I don't want to hard fail either. Code will continue to work as it was before, and happily ignore the `src_lang` and `tgt_lang` here.

Note: The validation already exists for `pipeline` function : https://github.com/huggingface/transformers/blob/master/src/transformers/pipelines/__init__.py#L234

Note: Your comment led me to do some testing, and I found a bug within my implementation for m2m100 tokenizer.. Thanks.

Be